### ISSUE TITLE
Get VSAN Health

### ISSUE THREAD

**Issue by jjahns** [dnaeon/py-vpoller#141]
Is there a way using this to get the properties for VSAN from a host?

The managed object path is
host->configManager->vsanSystem

There's probably a better way to get this information; however, it requires running a method on this object:

QueryHostStatus

This will return some data, namely the "health" value which indicates the VSAN health for that particular host.  This value is important, because if VSAN is unhealthy, the host will not get populated with a warning or error.

Unhealthy VSAN means that a drive failed or there is some lingering issue.  For me its critical for monitoring.  This is the only thing holding me back from plunging head first into this solution.

**Comment by dnaeon** [comment]
Hey,

I've just committed a new vPoller task for retrieving the VSAN health state for a host - the new task is called `vsan.health.get`.

Here is how to get the VSAN health for a host from the command-line for example.

```bash
$ vpoller-client --method vsan.health.get --vsphere-host vc01.example.org --name esxi01.example.org
```

Make sure that you update vPoller to version 0.5.8 in order to make use of the new VSAN task.

Let me know how it goes when you have time to test it out.

Thanks,
Marin

**Comment by jjahns** [comment]
There are a couple of additional items to grab as well:

nodeUuid - grabs the node UUID for VSAN
uuid - grabs the UUID for VSAN

Also, I found one problem - if the host is in a poweredOff or notResponding / disconnected state, this method will generate a vmodl.fault because it cannot query the host, so a check might need to exist for this.

What I've identified is that 2 checks need to exist - one for connectionState (connected), and one for powerState (poweredOn).

So if we can put in that and allow the other 2 attributes to be polled, then I think we've accounted for everything that needs it.
